lib/Sema/TypeCheckStmt.cpp

```diff
@@ -1644,7 +1635,9 @@ static Type getFunctionBuilderType(FuncDecl *FD) {
 }
 
 bool TypeChecker::typeCheckAbstractFunctionBody(AbstractFunctionDecl *AFD) {
-  auto res = TypeChecker::typeCheckAbstractFunctionBodyUntil(AFD, SourceLoc());
+  auto res =
+      evaluateOrDefault(AFD->getASTContext().evaluator,
+                        TypeCheckFunctionBodyRequest{AFD, SourceLoc()}, true);
```

Comment:
Am I right in assuming that `SourceLoc()` means to typecheck the whole body? Perhaps it deserves a brief comment on the request class.